### PR TITLE
Mobile Theme: remove feature

### LINKED ISSUE #2061: Minileven: can't load custom theme with jetpack_mobile_template or jetpack_mobile_stylesheet
Jetpack's Mobile Theme detection includes 2 filters, `jetpack_mobile_template` and `jetpack_mobile_stylesheet`, that allow you to define a custom Mobile Theme instead of the default Minileven:
https://github.com/Automattic/jetpack/blob/77629b6de0a3651b5cff0e6139682c0e8d388369/modules/minileven/minileven.php#L84

Unfortunately, the custom theme still has to be located inside the Minileven module directory. It would be nice if you could use the filters to load a theme from the /themes directory, or from another plugin.

Reported here:
https://wordpress.org/support/topic/custom-mobile-theme-1

### LINKED ISSUE #5815: Custom CSS / Mobile Theme: CSS is not added to minileven
#### Steps to reproduce the issue

0. Start with a site using WP 4.7, JP 4.4.2, and the Mobile Theme is enabled.
1. Go to Appearance > Edit CSS.
2. Add some custom CSS.
3. Save your changes.
4. Notice the CSS loaded on your site.
5. Load your site on mobile. Jetpack's Mobile Theme is loaded.
6. The Custom CSS isn't loaded anywhere on mobile.

Although Minileven calls `wp_head`, I assume there might be a problem with the way we define the active theme with the `option_stylesheet` and `option_template` filters.

This is a big issue for all site owners who previously used Jetpack's Custom CSS module to customize the look of the Mobile Theme, thanks to the option that was provided in the Custom CSS editor to load Custom CSS on mobile as well.
Right now, the only options they have to customize that mobile theme is to edit core plugin files or create their own plugin to load a custom stylesheet.

Until this gets fixed, a quick fix is to install this little plugin:
https://github.com/jeherve/jetpack-addons/archive/mobile-custom-css.zip

Reported here:
https://wordpress.org/support/topic/site-title-color-on-jetpack-mobile-changed-after-update-it-is-not-invisible/

### LINKED ISSUE #6616: Mobile theme: Enable Infinite Scroll on mobile theme
Can we enable infinite scroll on the mobile theme?

Requested in 3092923-t

### LINKED ISSUE #6617: Mobile theme: Share by email form position on mobile
#### Steps to reproduce the issue

- Enable mobile theme and sharing by email.
- View a post on mobile and tap the share by email icon.

#### What I expected

The form should display within the screen boundary and should not affect the rest of the page.

#### What happened instead

On Android (Google Pixel), the form reduces the size of the rest of the page in order to fit on the screen:

image

On iOS, it seems that the form just overlaps the boundaries of the screen, requiring horizontal scrolling:

image

Reported in 3092923-t

### LINKED ISSUE #10685: Jetpack Mobile Theme does not display subcategories
#### Steps to reproduce the issue

1. Set your site to display subcategories in WooCommerce. (Appearance > Customize > WooCommerce > Product Catalog)
2. Set "Enable Jetpack Mobile Theme"
3. Create a subcategory. (Products > Categories)
4. View parent category page on mobile.

#### What I expected

Subcategory to be visible on both regular browser and mobile.

#### What happened instead

zd-1574971

Subcategory was visible on regular site. Only products visible on mobile. Verified on my personal site.

**Screenshots**

image

image

### LINKED ISSUE #13385: Retire mobile theme (minileven)
With the majority of themes now responsive we should consider the retirement of the mobile theme feature. Ideally it will be off by default and then we can work towards a plan for retirement.

p2EDhh-Ia-p2
p1HpG7-7l5-p2

*****

**Project**: 1132000191628751-as-1150873576828085

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #13385

Closes #6616 
Closes #5815
Closes #6617
Closes #10685
Closes #2061

#### Changes proposed in this Pull Request:

It is time.

We originally created the mobile theme feature as a fall-back when the regular theme did not include a mobile view. Most themes include a mobile view by default now, so the feature is no longer necessary.

This PR is the last step in deprecating the Mobile Theme feature. In #14101 (as well as messages in the dashboard and emails), we started warning folks that the feature was going away. In this PR, we remove the feature entirely:

- We deactivate the module if it still active.
- We deactivate the module on plugin update.
- We delete options created by the module to clean things up.
- We remove the UI from the Jetpack Settings dashboard.
- We output notices to folks calling functions that were used by the module, or calling the module file directly.
- We stop synchronizing options with WordPress.com.
- We output notices to all folks who may still use filters / hooks to customize the mobile theme.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* It removes an existing feature. Internal reference: p1HpG7-85q-p2

#### Testing instructions:

* Start from a site that does not run this branch.
* Run `wp jetpack module activate minileven` to activate the Mobile Theme. 
* Switch to this branch.
* The Mobile Theme should not be active when you visit your site on a mobile device.
* You should see no PHP errors.

#### Proposed changelog entry for your changes:

* Mobile Theme: sunset feature. We originally created the mobile theme feature as a fall-back when the regular theme did not include a mobile view. Most themes include a mobile view by default now, so the feature is no longer necessary.

### CODE REVIEW

File: class.jetpack-modules-list-table.php [hunk 1]
```diff
@@ -96,9 +96,7 @@ function js_templates() {
 			<# var i = 0;
 			if ( data.items.length ) {
 			_.each( data.items, function( item, key, list ) {
-				if ( item === undefined ) return;
-				if ( 'minileven' == item.module && ! item.activated ) return;
-				if ( 'manage' == item.module && item.activated ) return; #>
```
Comment (jeherve) [inline class.jetpack-modules-list-table.php]: I took the opportunity to delete this reference to the Manage module.

File: packages/sync/src/class-defaults.php [hunk 2]
```diff
@@ -395,7 +392,7 @@ public static function get_callable_whitelist() {
 		'snitch',
 		'vip-legacy-redirect',
 		'wp_automatic',
-		'wp-rest-api-log', // https://wordpress.org/plugins/wp-rest-api-log/
+		'wp-rest-api-log', // https://wordpress.org/plugins/wp-rest-api-log/.
```
Comment (jeherve) [inline packages/sync/src/class-defaults.php]: This was needed to avoid linting errors.

**Comment by kraftbj** [comment]
Reads well. Going to spin up a JN to poke around a bit.

**Review by kraftbj — APPROVED** [review]
Noting that this notice in Calypso needs to be removed sooner rather than later:

2020-02-18 at 8 54 AM

But functionally, it doesn't allow me to activate the mobile theme, so that's not a blocker for the PR.

Otherwise, I see no mention of the mobile theme nor errors. 🚢

**Comment by jeherve** [comment]
> Noting that this notice in Calypso needs to be removed sooner rather than later

Yup, that's next on my list, and will be gone for anyone running Jetpack 8.3+.

**Pull request merged**